### PR TITLE
Refactored Errors Out, Signup & Login Flow Set Up

### PR DESCRIPTION
**Pull request by MartyTheeMartian**

### CODE REVIEW

File: src/components/NavBar/LogIn/LogIn.js [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+      document.getElementById('input-username-login').value = username;
+      document.getElementById('remember-username').checked = 'checked';
+    }
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    let username = document.getElementById('input-username-login').value;
+    let password = document.getElementById('input-password-login').value;
+    if(document.getElementById('remember-username').checked) {
```
Comment (elimisteve) [inline src/components/NavBar/LogIn/LogIn.js:18]: @MartyTheeMartian `const` rather than `let`, yes?

File: src/components/NavBar/NavBar.js [hunk 2]
```diff
@@ -17,11 +17,11 @@ class NavBar extends Component {
...
         <Navbar.Collapse>
-          <Nav>
-            <NavItem>
+          <ul className="nav navbar-nav">
+            <li role="presentation">
               <Link to="/dashboard">Dashboard</Link>
-            </NavItem>
-          </Nav>
+            </li>
+          </ul>
```
Comment (elimisteve) [inline src/components/NavBar/NavBar.js:24]: @MartyTheeMartian Was this to solve the nested `<a>` tag warning?

Comment (MartyTheeMartian) [reply src/components/NavBar/NavBar.js:24]: @elimisteve Yes, <NavItem> component comes with an `<a>` tag. I just inspected the component, and mimicked it's structure and replaced the `<a>` tag

File: src/components/NavBar/LogIn/LogIn.js (hunk 1 above)
Comment (MartyTheeMartian) [reply src/components/NavBar/LogIn/LogIn.js:18]: @elimisteve I initially wrote those lines with `const` but I wasn't sure if we were going to reuse those variables after they've been encrypted and returned.

**Comment by elimisteve** [comment]
@MartyTheeMartian Creating a new one?

**Comment by MartyTheeMartian** [comment]
@elimisteve Yes, I realized that the code can be more "cleaner" by using React's `ref` attribute as Travis had suggested. I've seen a lot of code examples of other devs storing username and password in the component's state, and dynamically setting it with the `onChange` attribute or even storing it in Redux. This would be very useful for most modern apps with autocomplete or autosuggest on key presses. Another similar method would be storing it in the class instance's properties, such as declaring `this.username` & `this.password` in the constructor. This is actually more clean than the previous method because each time state is set, `render()` is called .... causing an unnecessary re-render. But for our case, we don't want to leave anything floating in RAM, especially an un-hashed password!! So, both of these methods are out the window. I'll refactor using React's `ref` attribute because it's better practice to manipulate the React Virtual DOM through react's engine, rather than directly manipulating the actual DOM (which React won't know we made changes to ... ).

**Comment by elimisteve** [comment]
@MartyTheeMartian Sounds good.  Grabbing a value directly from the DOM should be fine, but yeah we should be able to avoid manipulating the DOM without telling React.

**Pull request closed without merging**